x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/literal/interval/Intervals.java

```diff
@@ -341,6 +341,10 @@ TemporalAmount parse(Source source, String string) {
                                             + ": negative value [{}] not allowed (negate the entire interval instead)",
                                     v);
                         }
+                        if (units.get(unitIndex) == TimeUnit.MILLISECOND && number.length() < 3) {
+                            // normalize the number past DOT to millis
+                            v *= number.length() < 2 ? 100 : 10;
```

Comment:
Same concept here, you could avoid the multiplication by creating a string with right zero padding and then read is as number.

Reply:
This might be changed later if we ever bring the nanos to intervals and a multiplication might be more succinct. But yes, also here debatable.